### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by areinsvo**
- src directory has been moved to backup/src
- src_complete directory is now called src
- For the p2z time, we are now using the C++ high resolution timer
- All versions should have the same format for the timing printouts
- Removed CudaEvent from the cuda versions. If we want to measure computation and memory transfer times separately, we will rely on profiling instead of timers.

For the record, these variables should have the same defaults in all implementations but can be changed from the makefile: 
- NITER=5 
- nlayer=20
- nevts=100
- ntracks=9600
- nb=ntracks/bsize

These variables can be optimized per implementation:
- bsize 
- nthreads
- nstreams
- nblocks

### CODE REVIEW

File: src/propagate-toz-test_CUDA.cu [hunk 1]
```diff
@@ -492,117 +637,90 @@ int main (int argc, char* argv[]) {
...
-   printf("done ntracks=%i tot time=%f (s) time/trk=%e (s)\n", nevts*ntrks*int(NITER), (elapsedtime)*0.001, (elapsedtime)*0.001/(nevts*ntrks));
-   printf("data region time=%f (s)\n", regiontime*0.001);
-   printf("memory transfer time=%f (s)\n", (copytime+copybacktime)*0.001);
-   printf("setup time time=%f (s)\n", (end_setup-start_setup)*0.001);
-   printf("formatted %i %i %i %i %i %f %f %f %f %i\n",int(NITER),nevts,ntrks, bsize,nb, (elapsedtime)*0.001, (regiontime)*0.001,  (copytime+copybacktime)*0.001, (end_setup-start_setup)*0.001, num_streams);
+   auto wall_diff = wall_stop - wall_start;
+   auto wall_time = static_cast<double>(std::chrono::duration_cast<std::chrono::microseconds>(wall_diff).count()) / 1e6;
+   printf("setup time time=%f (s)\n", (setup_stop-setup_start)*0.001);
+   printf("done ntracks=%i tot time=%f (s) time/trk=%e (s)\n", nevts*ntrks*int(NITER), wall_time, wall_time/(nevts*ntrks*int(NITER)));
+   printf("formatted %i %i %i %i %i %f 0 %f %i\n",int(NITER),nevts, ntrks, bsize, nb, wall_time, (setup_stop-setup_start)*0.001, nthreads);
```
Comment (tresreid) [inline src/propagate-toz-test_CUDA.cu]: change nthreads to num_streams

File: src/propagate-toz-test_CUDA.cu [hunk 2]
```diff
@@ -492,117 +637,90 @@ int main (int argc, char* argv[]) {
...
+  setup_stop = (long)timecheck.tv_sec * 1000 + (long)timecheck.tv_usec / 1000;
 
   printf("done preparing!\n");
-  //long start, end;
-  //long start2, end2;
-  //struct timeval timecheck;
 
   printf("Size of struct MPTRK trk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
   printf("Size of struct MPTRK outtrk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
   printf("Size of struct struct MPHIT hit[] = %ld\n", nevts*nb*sizeof(struct MPHIT));
```
Comment (tresreid) [inline src/propagate-toz-test_CUDA.cu:653]: add wall start time

File: Makefile [hunk 3]
```diff
@@ -247,10 +247,11 @@ endif
 TARGET = ./bin
 BENCHMARK = "propagate_$(COMPILER)_$(MODE)"
 
+CFLAGS1 += -std=c++11
```
Comment (tresreid) [inline Makefile]: alpaka doesn't like this. std=c++17 works

**Review by tresreid — APPROVED** [review]

**Comment by areinsvo** [comment]
@mastino I think the kokkos version should be okay now, but you'll have to test it and let me know if I broke anything. For now I'll go ahead and merge this PR.

@seyonglee note that I applied these changes to the OpenMPv4 version as well to keep everything consistent.

**Pull request merged**